internal/util.go

```diff
@@ -32,3 +39,70 @@ func RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]i
 
 	return
 }
+
+// PrepareTLSConfig generates TLS config based on the specifed parameters
+func PrepareTLSConfig(caCertFile, clientCertFile, clientKeyFile string, insecure *bool) (*tls.Config, error) {
+	config := &tls.Config{}
+	if caCertFile != "" {
+		caCert, _, err := pathOrContents(caCertFile)
+		if err != nil {
+			return nil, fmt.Errorf("Error reading CA Cert: %s", err)
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM([]byte(caCert))
+		config.RootCAs = caCertPool
+	}
+
+	if insecure == nil {
+		config.InsecureSkipVerify = false
+	} else {
+		config.InsecureSkipVerify = *insecure
+	}
+
+	if clientCertFile != "" && clientKeyFile != "" {
+		clientCert, _, err := pathOrContents(clientCertFile)
+		if err != nil {
+			return nil, fmt.Errorf("Error reading Client Cert: %s", err)
+		}
+		clientKey, _, err := pathOrContents(clientKeyFile)
+		if err != nil {
+			return nil, fmt.Errorf("Error reading Client Key: %s", err)
+		}
+
+		cert, err := tls.X509KeyPair([]byte(clientCert), []byte(clientKey))
+		if err != nil {
+			return nil, err
+		}
+
+		config.Certificates = []tls.Certificate{cert}
+		config.BuildNameToCertificate()
+	}
+
+	return config, nil
+}
+
+func pathOrContents(poc string) (string, bool, error) {
+	if len(poc) == 0 {
+		return poc, false, nil
+	}
+
+	path := poc
+	if path[0] == '~' {
+		var err error
+		path, err = homedir.Expand(path)
+		if err != nil {
+			return path, true, err
+		}
+	}
+
+	if _, err := os.Stat(path); err == nil {
+		contents, err := ioutil.ReadFile(path)
+		if err != nil {
+			return string(contents), true, err
```

Comment:
can you return `[]byte` in this func? this will allow to avoid excessive `string` -> `[]byte` conversion

Reply:
yes, sure! this conversion is absolutely unnecessary